No more authentication methods available

### Plugin Modernizer version

plugin modernizer 999999-SNAPSHOT (2025-01-20T10:11:56Z)

### What Operating System are you using

Debian on WSL2

### Reproduction steps

```
java -jar plugin-modernizer-cli/target/jenkins-plugin-modernizer-999999-SNAPSHOT.jar run --plugin-file plugins.txt --recipe SetupJenkinsfile
Run Plugin Modernizer
Skipping verification for plugin absint-a3
Forking plugin absint-a3 locally from repo absint-a3-plugin...
Synced the forked repository for plugin absint-a3
Fetching plugin code locally absint-a3...
Failed to fetch the repository
org.eclipse.jgit.api.errors.TransportException: Unable to connect
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:249)
        at io.jenkins.tools.pluginmodernizer.core.github.GHService.fetchRepository(GHService.java:598)
        at io.jenkins.tools.pluginmodernizer.core.github.GHService.fetch(GHService.java:543)
        at io.jenkins.tools.pluginmodernizer.core.model.Plugin.fetch(Plugin.java:765)
        at io.jenkins.tools.pluginmodernizer.core.impl.PluginModernizer.process(PluginModernizer.java:209)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at io.jenkins.tools.pluginmodernizer.core.impl.PluginModernizer.start(PluginModernizer.java:168)
        at io.jenkins.tools.pluginmodernizer.cli.command.RunCommand.call(RunCommand.java:99)
        at io.jenkins.tools.pluginmodernizer.cli.command.RunCommand.call(RunCommand.java:19)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at io.jenkins.tools.pluginmodernizer.cli.Main.main(Main.java:50)
Caused by: org.eclipse.jgit.errors.TransportException: Unable to connect
        at org.apache.sshd.git.transport.GitSshdSessionFactory.getSession(GitSshdSessionFactory.java:134)
        at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:107)
        at org.eclipse.jgit.transport.TransportGitSsh$SshFetchConnection.<init>(TransportGitSsh.java:285)
        at org.eclipse.jgit.transport.TransportGitSsh.openFetch(TransportGitSsh.java:152)
        at org.eclipse.jgit.transport.FetchProcess.executeImp(FetchProcess.java:153)
        at org.eclipse.jgit.transport.FetchProcess.execute(FetchProcess.java:105)
        at org.eclipse.jgit.transport.Transport.fetch(Transport.java:1458)
        at org.eclipse.jgit.api.FetchCommand.call(FetchCommand.java:238)
        ... 17 common frames omitted
Caused by: org.apache.sshd.common.SshException: No more authentication methods available
        at org.apache.sshd.common.future.AbstractSshFuture.verifyResult(AbstractSshFuture.java:141)
        at org.apache.sshd.client.future.DefaultAuthFuture.verify(DefaultAuthFuture.java:56)
        at org.apache.sshd.client.future.DefaultAuthFuture.verify(DefaultAuthFuture.java:35)
        at org.apache.sshd.common.future.VerifiableFuture.verify(VerifiableFuture.java:110)
        at org.apache.sshd.common.future.VerifiableFuture.verify(VerifiableFuture.java:96)
        at org.apache.sshd.git.transport.GitSshdSession.createClientSession(GitSshdSession.java:112)
        at org.apache.sshd.git.transport.GitSshdSessionFactory$1.createClientSession(GitSshdSessionFactory.java:110)
        at org.apache.sshd.git.transport.GitSshdSession.<init>(GitSshdSession.java:73)
        at org.apache.sshd.git.transport.GitSshdSessionFactory$1.<init>(GitSshdSessionFactory.java:90)
        at org.apache.sshd.git.transport.GitSshdSessionFactory.getSession(GitSshdSessionFactory.java:90)
        ... 24 common frames omitted
Caused by: org.apache.sshd.common.SshException: No more authentication methods available
        at org.apache.sshd.client.session.ClientUserAuthService.tryNext(ClientUserAuthService.java:441)
        at org.apache.sshd.client.session.ClientUserAuthService.processUserAuth(ClientUserAuthService.java:375)
        at org.apache.sshd.client.session.ClientUserAuthService.process(ClientUserAuthService.java:270)
        at org.apache.sshd.common.session.helpers.CurrentService.process(CurrentService.java:109)
        at org.apache.sshd.common.session.helpers.AbstractSession.doHandleMessage(AbstractSession.java:625)
        at org.apache.sshd.common.session.helpers.AbstractSession.lambda$handleMessage$0(AbstractSession.java:546)
        at org.apache.sshd.common.util.threads.ThreadUtils.runAsInternal(ThreadUtils.java:68)
        at org.apache.sshd.common.session.helpers.AbstractSession.handleMessage(AbstractSession.java:545)
        at org.apache.sshd.common.session.helpers.AbstractSession.decode(AbstractSession.java:1729)
        at org.apache.sshd.common.session.helpers.AbstractSession.messageReceived(AbstractSession.java:506)
        at org.apache.sshd.common.session.helpers.AbstractSessionIoHandler.messageReceived(AbstractSessionIoHandler.java:64)
        at org.apache.sshd.common.io.nio2.Nio2Session.handleReadCycleCompletion(Nio2Session.java:409)
        at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:382)
        at org.apache.sshd.common.io.nio2.Nio2Session$1.onCompleted(Nio2Session.java:377)
        at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.lambda$completed$0(Nio2CompletionHandler.java:38)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
        at org.apache.sshd.common.io.nio2.Nio2CompletionHandler.completed(Nio2CompletionHandler.java:37)
        at java.base/sun.nio.ch.Invoker.invokeUnchecked(Invoker.java:129)
        at java.base/sun.nio.ch.Invoker$2.run(Invoker.java:221)
        at java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:113)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
Failed to fetch the repository
```

### Expected Results

Not an SSH/git error.

It does not work directly on my machine during a Powershell session either.
Would be great to have a test (integration?) to reproduce the issue on a Windows environment

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/github/GHService.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/github/GHService.java
@@ -188,6 +188,7 @@ public class GHService {
 
     /**
      * Refresh the JWT token for the GitHub app. Only for GitHub App authentication
+     *
      * @param installationId The installation ID
      */
     public void refreshToken(Long installationId) {
@@ -215,6 +216,7 @@ public class GHService {
 
     /**
      * Get the repository object for a plugin
+     *
      * @param plugin The plugin to get the repository for
      * @return The GHRepository object
      */
@@ -228,6 +230,7 @@ public class GHService {
 
     /**
      * Get a plugin repository to the organization or personal account
+     *
      * @param plugin The plugin
      * @return The GHRepository object
      */
@@ -244,6 +247,7 @@ public class GHService {
 
     /**
      * Check if the plugin repository is forked to the organization or personal account
+     *
      * @param plugin The plugin to check
      * @return True if the repository is forked
      */
@@ -264,6 +268,7 @@ public class GHService {
 
     /**
      * Check if the plugin repository is archived
+     *
      * @param plugin The plugin to check
      * @return True if the repository is archived
      */
@@ -276,6 +281,7 @@ public class GHService {
 
     /**
      * Fork a plugin repository to the organization or personal account
+     *
      * @param plugin The plugin to fork
      */
     public void fork(Plugin plugin) {
@@ -303,8 +309,9 @@ public class GHService {
 
     /**
      * Fork a plugin repository to the organization or personal account
+     *
      * @param plugin The plugin to fork
-     * @throws IOException Forking the repository failed due to I/O error
+     * @throws IOException          Forking the repository failed due to I/O error
      * @throws InterruptedException Forking the repository failed due to interruption
      */
     private GHRepository forkPlugin(Plugin plugin) throws IOException, InterruptedException {
@@ -339,10 +346,11 @@ public class GHService {
 
     /**
      * Fork the repository
+     *
      * @param originalRepo The original repository to fork
      * @param organization The organization to fork the repository to. Can be null for personal account
      * @return The forked repository
-     * @throws IOException If the fork operation failed
+     * @throws IOException          If the fork operation failed
      * @throws InterruptedException If the fork operation was interrupted
      */
     private GHRepository forkRepository(GHRepository originalRepo, GHOrganization organization)
@@ -360,9 +368,10 @@ public class GHService {
 
     /**
      * Fork the repository to the personal account
+     *
      * @param originalRepo The original repository to fork
      * @return The forked repository
-     * @throws IOException If the fork operation failed
+     * @throws IOException          If the fork operation failed
      * @throws InterruptedException If the fork operation was interrupted
      */
     private GHRepository forkRepository(GHRepository originalRepo) throws IOException, InterruptedException {
@@ -371,6 +380,7 @@ public class GHService {
 
     /**
      * Get the organization object for the given owner or null if the owner is not an organization
+     *
      * @return The GHOrganization object or null
      * @throws IOException If the organization access failed
      */
@@ -385,8 +395,9 @@ public class GHService {
 
     /**
      * Check if the repository is forked on the given organization
+     *
      * @param organization The organization to check
-     * @param repoName The name of the repository
+     * @param repoName     The name of the repository
      * @return True if the repository is forked
      * @throws IOException If the repository access failed
      */
@@ -399,8 +410,9 @@ public class GHService {
 
     /**
      * Get the forked repository on the given organization
+     *
      * @param organization The organization to check
-     * @param repoName The name of the repository
+     * @param repoName     The name of the repository
      * @return The forked repository
      * @throws IOException If the repository access failed
      */
@@ -410,6 +422,7 @@ public class GHService {
 
     /**
      * Return if the repository is forked on current access
+     *
      * @param repoName The name of the repository
      * @return True if the repository is forked
      * @throws IOException If the repository access failed
@@ -420,6 +433,7 @@ public class GHService {
 
     /**
      * Get the forked repository on the personal account
+     *
      * @param repoName The name of the repository
      * @return The forked repository
      * @throws IOException If the repository access failed
@@ -430,6 +444,7 @@ public class GHService {
 
     /**
      * Sync a fork repository from its original upstream. Only the main branch is synced in case multiple branches exist.
+     *
      * @param plugin The plugin to sync
      */
     public void sync(Plugin plugin) {
@@ -460,6 +475,7 @@ public class GHService {
 
     /**
      * Sync a fork repository from its original upstream. Only the main branch is synced in case multiple branches exist.
+     *
      * @param forkedRepo Forked repository
      * @throws IOException if an error occurs while syncing the repository
      */
@@ -470,6 +486,7 @@ public class GHService {
 
     /**
      * Delete a plugin repository fork to the organization or personal account
+     *
      * @param plugin The plugin of the fork to delete
      */
     public void deleteFork(Plugin plugin) {
@@ -519,6 +536,7 @@ public class GHService {
 
     /**
      * Fetch a plugin repository code from the fork or original repo in dry-run mode
+     *
      * @param plugin The plugin to fork
      */
     public void fetch(Plugin plugin) {
@@ -553,6 +571,7 @@ public class GHService {
 
     /**
      * Fetch the repository code into local directory of the plugin
+     *
      * @param plugin The plugin to fetch
      * @throws GitAPIException If the fetch operation failed
      */
@@ -619,25 +638,16 @@ public class GHService {
         }
         // Clone the repository
         else {
-            try (Git git = Git.cloneRepository()
-                    .setCredentialsProvider(getCredentialProvider())
-                    .setRemote("origin")
-                    .setURI(remoteUri.toString())
-                    .setDirectory(plugin.getLocalRepository().toFile())
-                    .call()) {
-                LOG.debug("Clone successfully from {}", remoteUri);
+            try {
+                cloneRepository(plugin, remoteUri);
             } catch (GitAPIException e) {
                 if (e.getCause() instanceof org.apache.sshd.common.SshException) {
                     LOG.warn("SSH authentication failed. Retrying with HTTPS...");
                     remoteUri = new URIish(repository.getHttpTransportUrl());
-                    try (Git gitHttps = Git.cloneRepository()
-                            .setCredentialsProvider(getCredentialProvider())
-                            .setRemote("origin")
-                            .setURI(remoteUri.toString())
-                            .setDirectory(plugin.getLocalRepository().toFile())
-                            .call()) {
-                        LOG.debug("Clone successfully from {}", remoteUri);
+                    try {
+                        cloneRepository(plugin, remoteUri);
                     } catch (GitAPIException ex) {
+                        LOG.error("HTTPS clone failed: {}", ex.getMessage());
                         plugin.addError("Failed to fetch the repository using HTTPS", ex);
                         plugin.raiseLastError();
                     }
@@ -648,8 +658,20 @@ public class GHService {
         }
     }
 
+    private void cloneRepository(Plugin plugin, URIish remoteUri) throws GitAPIException {
+        try (Git git = Git.cloneRepository()
+                .setCredentialsProvider(getCredentialProvider())
+                .setRemote("origin")
+                .setURI(remoteUri.toString())
+                .setDirectory(plugin.getLocalRepository().toFile())
+                .call()) {
+            LOG.debug("Clone successfully from {}", remoteUri);
+        }
+    }
+
     /**
      * Checkout the branch for the plugin. Creates the branch if not exists
+     *
      * @param plugin The plugin to checkout branch for
      */
     public void checkoutBranch(Plugin plugin) {
@@ -681,6 +703,7 @@ public class GHService {
 
     /**
      * Commit all changes in the plugin directory
+     *
      * @param plugin The plugin to commit changes for
      */
     public void commitChanges(Plugin plugin) {
@@ -736,6 +759,7 @@ public class GHService {
 
     /**
      * Sign the commit using SSH key. Set not sign if using GH_TOKEN
+     *
      * @param commit The commit command to sign
      * @return The signed commit command
      * @throws IOException If the SSH key reading failed
@@ -755,6 +779,7 @@ public class GHService {
 
     /**
      * Get the current user
+     *
      * @return The current user
      */
     public GHUser getCurrentUser() {
@@ -781,6 +806,7 @@ public class GHService {
 
     /**
      * Get the primary email of the user
+     *
      * @param user The user to get the primary email for
      * @return The primary email
      */
@@ -802,6 +828,7 @@ public class GHService {
 
     /**
      * Push the changes to the forked repository
+     *
      * @param plugin The plugin to push changes for
      */
     public void pushChanges(Plugin plugin) {
@@ -851,6 +878,7 @@ public class GHService {
 
     /**
      * Open a pull request for the plugin
+     *
      * @param plugin The plugin to open a pull request for
      */
     public void openPullRequest(Plugin plugin) {
@@ -922,6 +950,7 @@ public class GHService {
 
     /**
      * Get the current credentials provider
+     *
      * @return The credentials provider
      */
     private CredentialsProvider getCredentialProvider() {
@@ -933,6 +962,7 @@ public class GHService {
     /**
      * Return if the given repository has any pull request originating from it
      * Typically to avoid deleting fork with open pull requests
+     *
      * @param plugin The plugin to check
      * @return True if the repository has any pull request
      */
@@ -963,6 +993,7 @@ public class GHService {
 
     /**
      * Check if a pull request already exists for the branch to the target repo
+     *
      * @param plugin The plugin to check
      * @return The pull request if it exists
      */
@@ -986,6 +1017,7 @@ public class GHService {
 
     /**
      * Determine the GitHub owner from config or using current token
+     *
      * @return The GitHub owner
      */
     public String getGithubOwner() {
@@ -996,6 +1028,7 @@ public class GHService {
 
     /**
      * Return if SSH auth is used
+     *
      * @return True if SSH key is used
      */
     public boolean isSshKeyAuth() {
@@ -1004,8 +1037,9 @@ public class GHService {
 
     /**
      * Ensure the forked reository correspond of the origin parent repository
+     *
      * @param originalRepo The original repository
-     * @param fork The forked repository
+     * @param fork         The forked repository
      * @throws IOException If the check failed
      */
     private void checkSameParentRepository(Plugin plugin, GHRepository originalRepo, GHRepository fork)

--- a/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/github/GHService.java
+++ b/plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/github/GHService.java
@@ -626,6 +626,24 @@ public class GHService {
                     .setDirectory(plugin.getLocalRepository().toFile())
                     .call()) {
                 LOG.debug("Clone successfully from {}", remoteUri);
+            } catch (GitAPIException e) {
+                if (e.getCause() instanceof org.apache.sshd.common.SshException) {
+                    LOG.warn("SSH authentication failed. Retrying with HTTPS...");
+                    remoteUri = new URIish(repository.getHttpTransportUrl());
+                    try (Git gitHttps = Git.cloneRepository()
+                            .setCredentialsProvider(getCredentialProvider())
+                            .setRemote("origin")
+                            .setURI(remoteUri.toString())
+                            .setDirectory(plugin.getLocalRepository().toFile())
+                            .call()) {
+                        LOG.debug("Clone successfully from {}", remoteUri);
+                    } catch (GitAPIException ex) {
+                        plugin.addError("Failed to fetch the repository using HTTPS", ex);
+                        plugin.raiseLastError();
+                    }
+                } else {
+                    throw e;
+                }
             }
         }
     }

